include/boost/geometry/strategies/cartesian/disjoint_segment_box_with_info.hpp

```diff
@@ -0,0 +1,110 @@
+// Boost.Geometry
+
+// Copyright (c) 2020, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_CARTESIAN_DISJOINT_SEGMENT_BOX_WITH_INFO_HPP
+#define BOOST_GEOMETRY_STRATEGIES_CARTESIAN_DISJOINT_SEGMENT_BOX_WITH_INFO_HPP
+
+#include <boost/geometry/core/assert.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/algorithms/detail/assign_box_corners.hpp>
+#include <boost/geometry/algorithms/detail/overlay/segment_as_subrange.hpp>
+#include <boost/geometry/geometries/segment.hpp>
+#include <boost/geometry/strategies/intersection.hpp>
+#include <boost/geometry/strategies/intersection_result.hpp>
+#include <boost/geometry/strategies/cartesian/distance_pythagoras.hpp>
+#include <boost/geometry/util/select_calculation_type.hpp>
+#include <boost/geometry/policies/relate/intersection_points.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace disjoint
+{
+
+struct cartesian_segment_box_with_info
+{
+    template
+    <
+            typename Segment,
+            typename BoxPoint,
+            typename Strategy,
+            typename T
+    >
+    inline void static apply(Segment const& segment,
+                             BoxPoint const& box_point1,
+                             BoxPoint const& box_point2,
+                             Strategy const& strategy,
+                             T& result)
```

Comment:
This looks more like an algorithm than a strategy.

Reply:
for cartesian maybe, but not for non-cartesian